### PR TITLE
chore: migrate verbose logging now with flowId (AR-1872)

### PR DESCRIPTION
**Pull request by yamilmedina**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

This adds some context to the current logging. To achieve this, we are using a TAG to correctly log the application user/flow:
I came up with this list, but we can be either more generic or more specific:

 - SYNC
 - EVENT_RECEIVER
 - CONVERSATIONS 
 - CONNECTIONS
 - MESSAGES
 - SEARCH
 - LOGIN
 - REGISTER
 - CLIENTS
 - CALLING

### Causes (Optional)

If we want to opt-in to have more detailed logs, we'll have to make some adjustments in calls to logs

### Solutions

Use the `withTag` function as a middle step to log, already provided by the Kermit lib

### Notes

Next PR's will come to adjust the rest of the logging levels (error, info, warn, debug). But first I want to validate this approach

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by alexandreferris — COMMENTED** [review]
Great work @yamilmedina !

I just have a question, will we be needing to add `.withFlowId(APPLICATION_FLOW)` to every log?

**Comment by yamilmedina** [comment]
> Great work @yamilmedina !
> 
> I just have a question, will we be needing to add `.withFlowId(APPLICATION_FLOW)` to every log?

Oh! Sadly yes. 

I was expecting to do something like an [MDC context ](https://www.slf4j.org/apidocs/org/slf4j/MDC.html#put(kotlin.String,kotlin.String)) here (we have it on `Thread` world, thanks to `ThreadLocal`), but here we can't since Coroutines are different, but most important we are doing logs outside coroutines

**Comment by vitorhugods** [comment]
> > Great work @yamilmedina !
> > I just have a question, will we be needing to add `.withFlowId(APPLICATION_FLOW)` to every log?
> 
> Oh! Sadly yes.
> 
> I was expecting to do something like an [MDC context ](https://www.slf4j.org/apidocs/org/slf4j/MDC.html#put(kotlin.String,kotlin.String)) here (we have it on `Thread` world, thanks to `ThreadLocal`), but here we can't since Coroutines are different, but most important we are doing logs outside coroutines

We could, maybe create a custom element and add it to the CoroutineContext.
`coroutineContext[LoggingInfo] = SomeLogStuff(data, etc)`


BUT, these informations would only be accessible if the logging methods were suspendable. Non-suspending stuff wouldn't be able to access the information in the `CoroutineContext`.

Food for thought tho.

**Comment by yamilmedina** [comment]
> > > Great work @yamilmedina !
> > > I just have a question, will we be needing to add `.withFlowId(APPLICATION_FLOW)` to every log?
> > 
> > 
> > Oh! Sadly yes.
> > I was expecting to do something like an [MDC context ](https://www.slf4j.org/apidocs/org/slf4j/MDC.html#put(kotlin.String,kotlin.String)) here (we have it on `Thread` world, thanks to `ThreadLocal`), but here we can't since Coroutines are different, but most important we are doing logs outside coroutines
> 
> We could, maybe create a custom element and add it to the CoroutineContext. `coroutineContext[LoggingInfo] = SomeLogStuff(data, etc)`
> 
> BUT, these informations would only be accessible if the logging methods were suspendable. Non-suspending stuff wouldn't be able to access the information in the `CoroutineContext`.
> 
> Food for thought tho.

Yup! We can't do it from non-suspended contexts 😓. So I went for this approach, trying to be more explicit when we are logging. 

I don't know if it makes sense to continue with this 😅 ? I think it is a step forward (verbose tough) but we can have better track. Do you, guys @alexandreferris / @vitorhugods  have more ideas there  🪄 ? Since continuing will require to modify other parts of the code

**Comment by alexandreferris** [comment]
> > > > Great work @yamilmedina !
> > > > I just have a question, will we be needing to add `.withFlowId(APPLICATION_FLOW)` to every log?
> > > 
> > > 
> > > Oh! Sadly yes.
> > > I was expecting to do something like an [MDC context ](https://www.slf4j.org/apidocs/org/slf4j/MDC.html#put(kotlin.String,kotlin.String)) here (we have it on `Thread` world, thanks to `ThreadLocal`), but here we can't since Coroutines are different, but most important we are doing logs outside coroutines
> > 
> > 
> > We could, maybe create a custom element and add it to the CoroutineContext. `coroutineContext[LoggingInfo] = SomeLogStuff(data, etc)`
> > BUT, these informations would only be accessible if the logging methods were suspendable. Non-suspending stuff wouldn't be able to access the information in the `CoroutineContext`.
> > Food for thought tho.
> 
> Yup! We can't do it from non-suspended contexts 😓. So I went for this approach, trying to be more explicit when we are logging.
> 
> I don't know if it makes sense to continue with this 😅 ? I think it is a step forward (verbose tough) but we can have better track. Do you, guys @alexandreferris / @vitorhugods have more ideas there 🪄 ? Since continuing will require to modify other parts of the code

I'm all up to having a way more verbose log, even though it might cost a bit more time as we need to add new code but I think that in the long run it will only benefit us.

And if in case we find something that helps better we can always change it again and again 😄

**Review by alexandreferris — APPROVED** [review]

File: logger/src/commonMain/kotlin/com/wire/kalium/logger/KaliumLogger.kt [hunk 1]
```diff
@@ -72,6 +72,9 @@ class KaliumLogger(config: Config, vararg logWriters: LogWriter = arrayOf()) {
 
     val severity = config.severity
 
+    @Suppress("unused")
+    fun withFlowId(flowId: ApplicationFlow) = kermitLogger.withTag(flowId.name.lowercase())
```
Comment (yamilmedina) [inline logger/src/commonMain/kotlin/com/wire/kalium/logger/KaliumLogger.kt:76]: `TODO`: change it to `FeatureId`

**Comment by vitorhugods** [comment]
> > > > > Great work @yamilmedina !
> > > > > I just have a question, will we be needing to add `.withFlowId(APPLICATION_FLOW)` to every log?
> > > > 
> > > > 
> > > > Oh! Sadly yes.
> > > > I was expecting to do something like an [MDC context ](https://www.slf4j.org/apidocs/org/slf4j/MDC.html#put(kotlin.String,kotlin.String)) here (we have it on `Thread` world, thanks to `ThreadLocal`), but here we can't since Coroutines are different, but most important we are doing logs outside coroutines
> > > 
> > > 
> > > We could, maybe create a custom element and add it to the CoroutineContext. `coroutineContext[LoggingInfo] = SomeLogStuff(data, etc)`
> > > BUT, these informations would only be accessible if the logging methods were suspendable. Non-suspending stuff wouldn't be able to access the information in the `CoroutineContext`.
> > > Food for thought tho.
> > 
> > 
> > Yup! We can't do it from non-suspended contexts 😓. So I went for this approach, trying to be more explicit when we are logging.
> > I don't know if it makes sense to continue with this 😅 ? I think it is a step forward (verbose tough) but we can have better track. Do you, guys @alexandreferris / @vitorhugods have more ideas there 🪄 ? Since continuing will require to modify other parts of the code
> 
> I'm all up to having a way more verbose log, even though it might cost a bit more time as we need to add new code but I think that in the long run it will only benefit us.
> 
> And if in case we find something that helps better we can always change it again and again 😄

Can't avoid it. It's a price we gotta pay :D

**Review by MohammedMokresh — APPROVED** [review]

**Pull request merged**